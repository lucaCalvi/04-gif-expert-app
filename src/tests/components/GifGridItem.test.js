import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests de compontente GifGridItem', () => {

    const title = 'The Simpsons';
    const url = 'https://localhost:8080/simpsons.jpg';

    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Comprueba que el componente funciona', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de tener un parrafo con el titulo', () => { 

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    });

    test('Debe de tener la imagen igual al url y alt de los props', () => { 
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('Debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');

        expect(div.hasClass('animate__fadeIn')).toBe(true);

    });

});