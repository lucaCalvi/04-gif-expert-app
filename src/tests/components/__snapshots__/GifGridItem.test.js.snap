// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de compontente GifGridItem Comprueba que el componente funciona 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="The Simpsons"
    src="https://localhost:8080/simpsons.jpg"
  />
  <p>
    The Simpsons
  </p>
</div>
`;
